spring-integration-ip/src/main/java/org/springframework/integration/ip/dsl/UdpOutboundChannelAdapterSpec.java

```diff
@@ -0,0 +1,124 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.dsl;
+
+import java.util.function.Function;
+
+import org.springframework.integration.dsl.MessageHandlerSpec;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.ip.udp.UnicastSendingMessageHandler;
+import org.springframework.messaging.Message;
+
+/**
+ * A {@link MessageHandlerSpec} for {@link UnicastSendingMessageHandler}s.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class UdpOutboundChannelAdapterSpec
+		extends MessageHandlerSpec<UdpOutboundChannelAdapterSpec, UnicastSendingMessageHandler> {
+
+	protected UdpOutboundChannelAdapterSpec() {
+		super();
+	}
+
+	UdpOutboundChannelAdapterSpec(String destinationExpression) {
+		this.target = new UnicastSendingMessageHandler(destinationExpression);
+	}
+
+	/**
+	 * @param timeout the timeout socket option.
+	 * @return the spec.
+	 * @see UnicastSendingMessageHandler#setSoTimeout(int)
+	 */
+	public UdpOutboundChannelAdapterSpec soTimeout(int timeout) {
+		this.target.setSoTimeout(timeout);
+		return _this();
+	}
+
+	/**
+	 * @param size the send buffer size socket option.
+	 * @return the spec.
+	 * @see UnicastSendingMessageHandler#setSoSendBufferSize(int)
+	 */
+	public UdpOutboundChannelAdapterSpec soSendBufferSize(int size) {
+		this.target.setSoSendBufferSize(size);
+		return _this();
+	}
+
+	/**
+	 * @param localAddress the local address.
+	 * @return the spec.
+	 * @see UnicastSendingMessageHandler#setLocalAddress(String)
+	 */
+	public UdpOutboundChannelAdapterSpec localAddress(String localAddress) {
+		this.target.setLocalAddress(localAddress);
+		return _this();
+	}
+
+	/**
+	 * @param lengthCheck the length check boolean.
+	 * @return the spec.
+	 * @see UnicastSendingMessageHandler#setLengthCheck(boolean)
+	 */
+	public UdpOutboundChannelAdapterSpec lengthCheck(boolean lengthCheck) {
+		this.target.setLengthCheck(lengthCheck);
+		return _this();
+	}
+
+	/**
+	 * @param size the receive buffer size socket option.
+	 * @return the spec.
+	 * @see UnicastSendingMessageHandler#setSoReceiveBufferSize(int)
+	 */
+	public UdpOutboundChannelAdapterSpec soReceiveBufferSize(int size) {
+		this.target.setSoReceiveBufferSize(size);
+		return _this();
+	}
+
+	/**
+	 * @param ackCounter the ack counter.
+	 * @return the spec.
+	 * @see UnicastSendingMessageHandler#setAckCounter(int)
+	 */
+	public UdpOutboundChannelAdapterSpec ackCounter(int ackCounter) {
+		this.target.setAckCounter(ackCounter);
+		return _this();
+	}
+
+	/**
+	 * @param socketExpressionFunction the socket expression.
+	 * @return the spec.
+	 * @see UnicastSendingMessageHandler#setSocketExpression(org.springframework.expression.Expression)
+	 */
+	public UdpOutboundChannelAdapterSpec socketExpression(Function<Message<?>, ?> socketExpressionFunction) {
```

Comment:
It's definitely `socketFunction()`.
The fact transformation to a `FunctionExpression` is out of end-user API.

Plus this must be like `Function<Message<?>, DatagramSocket>`, just because:
```
if (this.socketExpression != null) {
	socket = this.socketExpression.getValue(this.evaluationContext, message, DatagramSocket.class);
}
```
therefore we restrict the target code with particular type and that is much easier to write the final lambda.